### PR TITLE
:sparkles: `[platform]` Add platform user management

### PR DESCRIPTION
**Pull request by acabarbaye**
### Description

- Adding utilities with regards to platform user management

### Test Coverage

- [ ]  This change is covered by existing or additional automated tests.
- [ ]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [x]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: utils/platform/users.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		return err
+	}
+	found, _ := HasUser(username)
+	if found {
+		return nil
+	}
+	return ConvertUserGroupError(addUser(ctx, username, fullname, password))
+}
+
+// DeleteUser removes a user from the platform
```
Comment (joshjennings98) [inline utils/platform/users.go:39]: Can there be more of a distinction between when to use remove Vs delete user? If I was importing this library I wouldn't know which to use

**Review by joshjennings98 — DISMISSED** [review]
Just change the descriptions around delete and remove

File: utils/platform/users.go (hunk 1 above)
Comment (acabarbaye) [inline utils/platform/users.go:39]:
```suggestion
// DeleteUser removes a user from the platform when the user is specified using a `user.User` structure.
```

Comment (acabarbaye) [reply utils/platform/users.go]: tried to do so

**Pull request merged**